packages/core/ios/Sources/DdSdk.swift

```diff
@@ -33,21 +35,27 @@ class RNDdSdk: NSObject {
         self.mainDispatchQueue = mainDispatchQueue
         super.init()
     }
-
-
+    
     @objc(initialize:withResolver:withRejecter:)
     func initialize(configuration: NSDictionary, resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) -> Void {
         // Datadog SDK init needs to happen on the main thread: https://github.com/DataDog/dd-sdk-reactnative/issues/198
         self.mainDispatchQueue.async {
```

Comment:
Could you explain what would be the benefits of doing so?

Reply:
I noticed that you were referencing self in the closure, which may lead to a retain cycle.

This is probably fine because you're calling async on a main thread, but it's usually a good idea to define the reference to self within the closure.

https://www.avanderlee.com/swift/weak-self/#:~:text=Use%20a%20weak%20reference%20whenever,has%20been%20set%20during%20initialization.